internal/operands/common-instancetypes/reconcile.go

```diff
@@ -58,50 +97,121 @@ func fetchClusterResources[C clusterType](bundlePath string) ([]C, error) {
+		return nil, "", err
+	}
+	if err = fSys.WriteFile(filepath.Join(tmpDir.String(), "kustomization.yaml"), []byte(fmt.Sprintf("\nresources:\n  - %s", *URI))); err != nil {
+		return nil, "", err
+	}
+	return fSys, tmpDir.String(), nil
 }
 
-func WatchClusterTypes() []operands.WatchType {
-	return []operands.WatchType{
-		{Object: &instancetypev1alpha2.VirtualMachineClusterInstancetype{}, Crd: instancetypeapi.ClusterPluralResourceName, WatchFullObject: true},
-		{Object: &instancetypev1alpha2.VirtualMachineClusterPreference{}, Crd: instancetypeapi.ClusterPluralPreferenceResourceName, WatchFullObject: true},
+func generateResourcesFromURI(URI *string) (resmap.ResMap, error) {
+	fSys, tmpDirPath, err := generateKustomization(URI)
+	if err != nil {
+		return nil, err
 	}
+	k := krusty.MakeKustomizer(krusty.MakeDefaultOptions())
+	return k.Run(fSys, tmpDirPath)
 }
 
-func (c *commonInstancetypes) WatchClusterTypes() []operands.WatchType {
-	return WatchClusterTypes()
+func fetchResourcesFromURI(request *common.Request) ([]instancetypev1alpha2.VirtualMachineClusterInstancetype, []instancetypev1alpha2.VirtualMachineClusterPreference, error) {
+	resmap, err := generateResourcesFromURI(request.Instance.Spec.CommonInstancetypes.URI)
+	if err != nil {
+		request.Logger.Error(err, "failed to generate kustomize resources from URI")
+		return nil, nil, err
+	}
+
+	var (
+		virtualMachineClusterInstancetypes []instancetypev1alpha2.VirtualMachineClusterInstancetype
+		virtualMachineClusterPreferences   []instancetypev1alpha2.VirtualMachineClusterPreference
+	)
+
+	for _, r := range resmap.Resources() {
+		switch strings.ToLower(r.GetKind()) {
+		case instancetypeapi.ClusterSingularResourceName:
+			bytes, err := r.AsYAML()
+			if err != nil {
+				return nil, nil, err
+			}
+			bundle, err := decodeResources[instancetypev1alpha2.VirtualMachineClusterInstancetype](bytes)
+			if err != nil {
+				return nil, nil, err
+			}
+			virtualMachineClusterInstancetypes = append(virtualMachineClusterInstancetypes, bundle...)
+		case instancetypeapi.ClusterSingularPreferenceResourceName:
+			bytes, err := r.AsYAML()
+			if err != nil {
+				return nil, nil, err
+			}
+			bundle, err := decodeResources[instancetypev1alpha2.VirtualMachineClusterPreference](bytes)
+			if err != nil {
+				return nil, nil, err
+			}
+			virtualMachineClusterPreferences = append(virtualMachineClusterPreferences, bundle...)
+		}
+	}
+	return virtualMachineClusterInstancetypes, virtualMachineClusterPreferences, nil
 }
 
-func (c *commonInstancetypes) WatchTypes() []operands.WatchType {
-	return nil
+func (c *commonInstancetypes) reconcileFromURI(request *common.Request) ([]common.ReconcileResult, error) {
+	request.Logger.Info(fmt.Sprintf("Reconciling common-instancetypes from URI %s", *request.Instance.Spec.CommonInstancetypes.URI))
+
+	// Remove any existing resources if the URI is new or has changed
+	if c.resourceURI == "" || c.resourceURI != *request.Instance.Spec.CommonInstancetypes.URI {
```

Comment:
This is definitely wrong and will cleanup resources on restart, I'll look again tomorrow.